src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueueClient.cs

```diff
@@ -12,40 +13,46 @@ namespace Enable.Extensions.Queuing.AzureServiceBus.Internal
     // TODO Respect cancellationToken
     public class AzureServiceBusQueueClient : BaseQueueClient
     {
-        private readonly string _connectionString;
-        private readonly string _queueName;
-        private readonly MessageReceiver _messageReceiver;
-        private readonly MessageSender _messageSender;
+        private static readonly ConcurrentDictionary<int, AzureServiceBusQueue> _queues
+            = new ConcurrentDictionary<int, AzureServiceBusQueue>();
 
+        private readonly int _queueKey;
+        private readonly AzureServiceBusQueue _queue;
         private bool _disposed;
 
         public AzureServiceBusQueueClient(
             string connectionString,
             string queueName)
         {
-            _connectionString = connectionString;
-            _queueName = queueName;
+            _queueKey = $"{connectionString}:{queueName}".GetHashCode();
```

Comment:
Is there a risk here that different casing of either the connection string or the queue name will result in multiple connections being used? We should look to apply some normalisation to these strings before we use them as a key.

Reply:
Are connection strings and queue names case insensitive?

Reply:
Done.